source/extensions/stat_sinks/wasm/wasm_stat_sink_impl.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "envoy/common/platform.h"
+#include "envoy/local_info/local_info.h"
+#include "envoy/network/connection.h"
+#include "envoy/stats/histogram.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/sink.h"
+#include "envoy/stats/stats.h"
+#include "envoy/stats/tag.h"
+#include "envoy/thread_local/thread_local.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/macros.h"
+
+#include "extensions/common/wasm/wasm.h"
+#include "extensions/stat_sinks/well_known_names.h"
+
+#include "absl/strings/str_join.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+namespace Wasm {
+
+using Envoy::Extensions::Common::Wasm::WasmHandle;
+
+class WasmStatSink : public Stats::Sink {
```

Comment:
Please add unit/integration tests for this class.

Given all the compilation errors, apparently, it doesn't have any tests.